sw/otbn/crypto/rsa_verify_3072_consts_test.s

```diff
@@ -0,0 +1,170 @@
+/* Copyright lowRISC contributors. */
+/* Licensed under the Apache License, Version 2.0, see LICENSE for details. */
+/* SPDX-License-Identifier: Apache-2.0 */
+
+.section .text.start
+
+/**
+ * Standalone test for 3072 bit RSA Montgomery constant computation. 
+ *
+ * Copies the message to wide registers for comparison (starting at
+ * w0). See comment at the end of the file for expected values.
+ */
+run_rsa_verify_3072_consts_test:
+  /* Compute R^2 = (2^3072)^2 mod M */
+  jal      x1, compute_rr
+
+  /* Compute m0_inv = (- (M^-1)) mod 2^256 */
+  jal      x1, compute_m0_inv
+
+  /* Copy m0_inv to w0 */
+  li       x8, 0
+  la       x21, m0inv
+  bn.lid   x8++, 0(x21)
+
+  /* copy all limbs of R^2 to wide reg file */
+  la       x21, rr
+  loopi    12, 2
+    bn.lid   x8, 0(x21++)
+    addi     x8, x8, 1
```

Comment:
Sorry, I asked this question in [the wrong place](https://github.com/lowRISC/opentitan/pull/9422#discussion_r759313035) before. But why are we loading things from memory just before the ECALL? The register file contents aren't externally visible.

Reply:
It's a test file intended for the simulator, and register contents are dumped after the simulation, so it's easier to read them from there than DMEM. This is based on similar logic in other pure-OTBN test files, e.g. https://github.com/lowRISC/opentitan/blob/f700bf7a12a919c3db6fab25fb9d015e317b95cd/sw/otbn/crypto/rsa_verify_3072_test.s#L20-L25

Reply:
Ahah! Makes sense, thanks.